plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -0,0 +1,419 @@
+/******************************************************************************
+* Copyright (c) 2019 TileDB, Inc.
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+
+#include <cctype>
+#include <limits>
+
+#include <boost/any.hpp>
+#include <pdal/util/FileUtils.hpp>
+
+#include "TileDBWriter.hpp"
+
+
+const char pathSeparator =
+#ifdef _WIN32
+        '\\';
+#else
+        '/';
+#endif
+
+namespace pdal {
+
+static PluginInfo const s_info
+{
+    "writers.tiledb",
+    "Write data using TileDB.",
+    "http://pdal.io/stages/drivers.tiledb.writer.html"
+};
+
+CREATE_SHARED_STAGE(TileDBWriter, s_info)
+
+template <typename T>
+void setAttrAnyValue(std::map<std::string, pdalboost::any>& mp, const std::string& key, const T val)
+{
+    auto it = mp.find(key);
+    if (it == mp.end())
+    {
+        auto sp = std::make_shared<std::vector<T>>();
+        mp[key] = sp;
+    }
+    auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<T>>>(mp[key]);
+    sp->push_back(val);
+}
+
+void writeAttributeValue(const std::string& attrName,
+        const PointViewPtr view, const PointId idx, std::map<std::string, pdalboost::any>& mp)
+{
+    Dimension::Id id = Dimension::id(attrName);
+    Dimension::Type t = view->dimType(id);
+
+    switch (t)
```

Comment:
I'm not sure exactly what you're trying to do here.  You don't need to switch on PDAL's type and cast.  PDAL will return whatever type your writer wants, without regard to the way the data is stored in PDAL.  If you want to write your data as doubles, just fetch them as doubles with "getFieldAs<double>(...)".  You don't need to worry about the internal type at all.

Reply:
I was aiming to write the data in the type it is stored in. I don't want to use more storage by assuming doubles.